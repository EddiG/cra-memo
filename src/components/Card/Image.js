// @flow
import { withProps } from 'recompose';
import styled from 'styled-components';
import NormalImage from 'components/Image';

const NonDraggableImage = withProps({ draggable: false })(NormalImage);

const Image = styled(NonDraggableImage)`
  position: absolute;
  display: block;
  width: 100%;
  backface-visibility: hidden;
  user-select: none;
`;

export default Image;
