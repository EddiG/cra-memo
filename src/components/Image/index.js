// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  src: string,
  alt: string,
  className: string,
};

const Image = ({ src, alt, className, ...props }: Props) => (
  <img src={src} alt={alt} className={className} {...props} />
);

const StyledImage = styled(Image)`

`;

export default StyledImage;
