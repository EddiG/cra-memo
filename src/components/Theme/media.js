import { css } from 'styled-components';

const baseFontSize = 16;

const sizes = {
  desktop: 1024,
  tablet: 768,
  phone: 376,
};

export default Object.keys(sizes).reduce((acc, target) => {
  const emSize = sizes[target] / baseFontSize;
  acc[target] = (...args) => css`
    @media (max-width: ${emSize}em) {
      ${css(...args)}
    }
  `;
  return acc;
}, {});
