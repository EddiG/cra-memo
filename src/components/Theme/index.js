// @flow
import React from 'react';
import { ThemeProvider } from 'styled-components';
import media from './media';

type Props = {
  children?: React.Children,
};

const theme = {
  colors: {
    fg: {
      main: '#b1b1b1',
      avatar: '#53bdff',
      button: '#fff',
    },
    bg: {
      nav: '#333333',
      main: '#fff',
      button: '#e0b305',
    },
  },
  media,
};

const Theme = ({ children }: Props) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

Theme.defaultProps = {
  children: null,
};

export default Theme;
