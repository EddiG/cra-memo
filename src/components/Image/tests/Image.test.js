import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../index';
import Theme from '../../Theme';

describe('Image component', () => {
  it('should rendered successfully', () => {
    const image = renderer.create(<Theme><Image /></Theme>).toJSON();
    expect(image).toMatchSnapshot();
  });
});
