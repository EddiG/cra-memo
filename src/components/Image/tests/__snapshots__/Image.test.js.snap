// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should rendered successfully 1`] = `
<img
  alt={undefined}
  className="sc-bdVaJa eAwGjB"
  src={undefined}
/>
`;
